src/uct/ib/dc/base/dc_iface.c

```diff
@@ -230,16 +229,13 @@ static ucs_status_t uct_dc_iface_init_version(uct_dc_iface_t *iface,
 
     if (ver & UCT_IB_DEVICE_FLAG_DC_V2) {
         iface->version_flag = UCT_DC_IFACE_ADDR_DC_V2;
-        return UCS_OK;
     }
 
     if (ver & UCT_IB_DEVICE_FLAG_DC_V1) {
         iface->version_flag = UCT_DC_IFACE_ADDR_DC_V1;
```

Comment:
I'm guessing, now it should be `|=` here and initialized with `0` in begin of the function. Otherwise, it will be always V1
Pls check workability on mixed configuration (CX4 + CX5)

Reply:
need to assert that not both flags are present